### PR TITLE
Popover: Base implementation using Floating-UI

### PR DESCRIPTION
**Pull request by diyorbek**
### Summary

Re-implemented core internal components of Popover using Floating-UI. 
New implementation is not exposed. This is the first PR of gradual rehaul of Popover and Popover based components.

The changes include:
 - New Contents component
 - New Controller component
 - New InternalPopover
 - `disablePortal` prop for InternalPopover
 - `scrollBoundary` prop for InternalPopover
 - `hideWhenReferenceHidden` prop for InternalPopover (open for shorter prop name suggestions)

### Links

- [Jira](https://jira.pinadmin.com/browse/GESTALT-6386)
- [TDD](https://paper.dropbox.com/doc/Gestalt-TDD-Popover-V2--CCkqLA2Iu~hjVD_9VnUXpK9pAQ-T6gxnHxXGWEJin2TkENXr)
- [Figma](link to Figma file)

### Checklist

- [ ] Added unit and Flow Tests
- [ ] Added documentation + accessibility tests
- [ ] Verified accessibility: keyboard & screen reader interaction
- [ ] Checked dark mode, responsiveness, and right-to-left support
- [ ] Checked stakeholder feedback (e.g. Gestalt designers)

### CODE REVIEW

File: docs/docs-components/DevelopmentEditor.js [hunk 1]
```diff
@@ -62,14 +62,16 @@ export default function DevelopmentEditor({ code }: { code: ?string | (() => Nod
...
-    .replace('export default', '')
+    .replace('export default', 'const App =')
     // Add React. to React imports
     .replace(reactRegex, 'React.$&')
     .replace(
       'const [showComponent, setShowComponent] = React.useState(true);',
       'const [showComponent, setShowComponent] = React.useState(false);',
     );
 
+  const codeWrapped = `function Root() {${codeFileCleaned || ''}; return <App />}`;
```
Comment (diyorbek) [inline docs/docs-components/DevelopmentEditor.js:73]: Just a small enhancement to make examples work

File: packages/gestalt/src/Popover/Contents.js [hunk 2]
```diff
@@ -0,0 +1,133 @@
+// @flow strict
```
Comment (diyorbek) [inline packages/gestalt/src/Popover/Contents.js:1]: Copied legacy Contents and converted to function component removing unecessary parts.

File: packages/gestalt/src/Popover/Controller.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
+// @flow strict
```
Comment (diyorbek) [inline packages/gestalt/src/Popover/Controller.js:1]: Copied legacy Controller and converted to function component removing unecessary parts.

File: packages/gestalt/src/Popover/usePopover.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  type Placement,
+  shift,
+  type Side,
+  type Strategy,
+  useFloating,
+  type UseFloatingReturn,
+} from '@floating-ui/react';
+import { type MainDirections } from '../utils/positioningTypes.js';
+
+export const DIRECTIONS_MAP: Record<MainDirections, Side | 'forceDown'> = {
```
Comment (diyorbek) [inline packages/gestalt/src/Popover/usePopover.js:18]: We need these for now to make our current `directions` work with Floating-UI's `placements`. After we roll out the chages we can remove this bridge and just use `placements` for.

File: packages/gestalt/src/Popover/Contents.js [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+    scrollBoundary,
+  });
+
+  const caretOffset = middlewareData.arrow;
+  const visibility = middlewareData.hide?.referenceHidden === true ? 'hidden' : 'visible';
+  const background = bgColor === 'white' ? `${bgColor}BgElevated` : `${bgColor}Bg`;
+  const bgColorElevated = bgColor === 'white' ? 'whiteElevated' : bgColor;
+  const isCaretVertical = placement === 'top' || placement === 'bottom';
+
+  const popoverCaret = (
```
Comment (AlbertCarreras) [inline packages/gestalt/src/Popover/Contents.js:64]: nit?
any benefit of keeping it here?
could we just inline this? more readable IMO inlined

File: packages/gestalt/src/Popover/Controller.js [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+        scrollBoundary={scrollBoundary}
+      >
+        {children}
+      </Contents>
+    </OutsideEventBehavior>
+  );
+
+  if (disablePortal) return contents;
+
+  return <FloatingPortal>{contents}</FloatingPortal>;
```
Comment (AlbertCarreras) [inline packages/gestalt/src/Popover/Controller.js:92]: nit

return disablePortal ? contents : <FloatingPortal>{contents}</FloatingPortal>;

File: packages/gestalt/src/Popover/usePopover.js [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+};
+
+export const SIDES_MAP: Record<Side, MainDirections> = {
+  'bottom': 'down',
+  'left': 'left',
+  'right': 'right',
+  'top': 'up',
+};
+
+interface Props {
```
Comment (AlbertCarreras) [inline packages/gestalt/src/Popover/usePopover.js:33]: can we add a description of each prop here?

 ```
 /**
   * description
   */
```

File: packages/gestalt/src/Popover/usePopover.js [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+  });
+
+  // #endregion
+
+  const floating = useFloating({
+    strategy,
+    placement,
+    whileElementsMounted: autoUpdate,
+    elements: { reference: anchor },
+    // Order of middlewares are important as the calculations are passed along
```
Comment (AlbertCarreras) [inline packages/gestalt/src/Popover/usePopover.js:83]: maybe add, Do not "reorder middleware"

File: packages/gestalt/src/Popover/Contents.js [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+  idealDirection?: MainDirections,
+  onKeyDown: (event: SyntheticKeyboardEvent<HTMLElement>) => void,
+  role: ?Role,
+  rounding?: 2 | 4,
+  shouldFocus?: boolean,
+  width: ?number,
+  scrollBoundary?: HTMLElement,
+};
+
+export default function Contents({
```
Comment (AlbertCarreras) [inline packages/gestalt/src/Popover/Contents.js:32]: Can we change name here or in File to  NewContents.

I know locations / paths are different but I want to make sure we dont exhange them by misatke. we cant rerename in the future

**Review by AlbertCarreras — APPROVED** [review]

File: packages/gestalt/src/Popover/Contents.js (hunk 9 above)
Comment (diyorbek) [reply packages/gestalt/src/Popover/Contents.js:32]: What about Experimental prefix? Or we can rename old ones with prefix Legacy.

Comment (AlbertCarreras) [reply packages/gestalt/src/Popover/Contents.js:32]: I'll leave it up to you. Legacy maybe makes more sense?

Comment (diyorbek) [reply packages/gestalt/src/Popover/Contents.js:33]: Renamed old one with `Legacy` prefix

**Pull request merged**